### DISCUSSION TITLE
[pg_cron] Job 1 does not exist or you don't own it

### DISCUSSION THREAD

**Discussion by MicrorealHQ** [supabase/supabase#22262] [Questions] [answered]
I am following this guide https://supabase.com/docs/guides/database/extensions/pg_cron#edit-a-job to update a job

I have created a job using this command:

```
select cron.schedule(
               'invoke-function-bulk-process-photos-every-x-minutes',
               '*/5 * * * *', -- every 15 minute
               $$
    select
      net.http_post(
          url:='https://endpoint',
          headers:='{"Content-Type": "application/json"}'::jsonb,
          body:=concat()::jsonb
      ) as request_id;
    $$
       );
```
Here is data in cron.job
1,*/5 * * * *,"
    select
      net.http_post(
         net.http_post(
          url:='https://endpoint',
          headers:='{"Content-Type": "application/json"}'::jsonb,
          body:=concat()::jsonb
      ) as request_id;
    ",localhost,5432,postgres,postgres,true,invoke-function-bulk-process-photos-every-x-minutes

Now I am trying to update this job from supabase edge function by calling this (supabase client .rpc using service_role)
```
SELECT cron.alter_job(
                   job_id := (select jobid from cron.job where jobname = input_jobname),
                   active := input_status
           );
```
And I got this error  `Job 1 does not exist or you don't own it`
When I tried to use Update cron.job Set, then I got other error `permission denied for table job`

Does anyone get this error?

Note: i tried to reinstall pg_cron and repeat these steps but still got error

**Comment by GaryAustin1** [comment, marked as answer]
Service_role user does not own cron jobs started Postgres role.  You might need to run the postgres function as security definer but then need to secure in a private schema or removing roles granted to it.

**Comment by MicrorealHQ** [comment]
Problem solved. Thanks @GaryAustin1
